project4/name_server/tasks/console_task.cpp

```diff
@@ -0,0 +1,37 @@
+#include "console_task.h"
+
+#include <loguru.hpp>
+#include <utility>
+
+namespace nameserver_tasks {
+
+ConsoleTask::ConsoleTask(const std::string& prompt) : prompt_(prompt) {}
+
+thread_pool::Task::Status ConsoleTask::SetUp() {
+  std::cout << prompt_ << std::flush;
+  return thread_pool::Task::Status::RUNNING;
+}
+
+thread_pool::Task::Status ConsoleTask::RunAtomic() {
+  const auto first_message = console_message_queue_.Pop();
```

Comment:
This should be retooled using `PopTimed`. If the timeout expires, it should simply return `Status::RUNNING`. That will save us the necessity of doing the ugly hack that I had to do to get this task to exit correctly in project 3.